src/main/java/mate/academy/bookstore/service/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,32 @@
+package mate.academy.bookstore.service.impl;
+
+import lombok.RequiredArgsConstructor;
+import mate.academy.bookstore.dto.user.UserRegistrationRequestDto;
+import mate.academy.bookstore.dto.user.UserResponseDto;
+import mate.academy.bookstore.exception.RegistrationException;
+import mate.academy.bookstore.mapper.UserMapper;
+import mate.academy.bookstore.model.User;
+import mate.academy.bookstore.repository.user.UserRepository;
+import mate.academy.bookstore.service.UserService;
+import org.springframework.stereotype.Component;
+
+@RequiredArgsConstructor
+@Component
+public class UserServiceImpl implements UserService {
+    private final UserRepository userRepository;
+    private final UserMapper userMapper;
+
+    @Override
+    public UserResponseDto save(UserRegistrationRequestDto requestDto)
+            throws RegistrationException {
+        String email = requestDto.getEmail().toLowerCase();
+        if (userRepository.findByEmail(email).isPresent()) {
```

Comment:
I'd also rather use this, because we do not need the actual data of that user, just the fact that it is present or not, which can be a bit faster than fetching it
```suggestion
        if (userRepository.existsByEmail(email)) {
```